### ISSUE TITLE
Exception caught: Should only call onDetach when the widget is attached to the browser's document

### ISSUE THREAD

**Issue by kyr0** [GwtMaterialDesign/gwt-material#460]
I'm getting an AttachDetachException "Exception caught: Should only call onDetach when the widget is attached to the browser's document" whenever I try to detach widgets after using MaterialDatePicker, MaterialTextBox etc.

I'm pre-checking if a widget is attached before trying to detach it. Like that:

    if (widget.isAttached()) {
         widget.detach();
    }

"widget" is a "Composite implements IsWidget" or an sub-class of it.

Looking for a solution I found (and several other comments that stated the same):
http://blog.adjective.org/post/2012/09/19/GWT-AttachDetachException

But there is no overriding of onAttach() or onDetach() in my codebase. 

But I found a place here:
https://github.com/GwtMaterialDesign/gwt-material/blob/69fe804f546b8fe296809e13b790259c701622a7/gwt-material/src/main/java/gwt/material/design/client/ui/html/Text.java

And there is no call to super.onAttach() or super.onDetach(). But the superclass Widget does implement onAttach() and onDetach().

Could this lead to the Exception I'm experiencing?

Thanks!

**Comment by kevzlou7979** [comment]
On 2.0 We have implemented the check of attaching / onload of widget and deattaching / unloading. The issue exists on errai because of CDI Injection which loaded twice. I think it's a similar issue but on 2.0 it will be supported. Correct me if it's connected to your issue.

**Comment by kyr0** [comment]
Thank you for the fast response. A quick status update from my side: Hmm yeah, I tried 2.0-SNAPSHOT, experienced the same issue; we're using plain GWT here, no Errai etc.; well, we've build our own micro-framework that covers the page soft-reload (DOM cleanup) logic. I switched the Composite's onDetach() and am calling Composite's removeFromParent() now. It seems to work now; but it feels like a quick fix hack -- I fear leaking event listeners and am not so sure about all implications. Have to check the GWT implementation of both methods now...
